official/vision/beta/projects/yolo/losses/yolo_loss.py

```diff
@@ -0,0 +1,666 @@
+      loss: `float` for the actual loss.
+      box_loss: `float` loss on the boxes used for metrics.
+      conf_loss: `float` loss on the confidence used for metrics.
+      class_loss: `float` loss on the classes used for metrics.
+      avg_iou: `float` metric for the average iou between predictions 
+        and ground truth.
+      avg_obj: `float` metric for the average confidence of the model 
+        for predictions.
+    """
+    (loss, box_loss, conf_loss, class_loss, mean_loss, iou, pred_conf, ind_mask,
+     grid_mask) = self._compute_loss(true_counts, inds, y_true, boxes, classes, y_pred)
+
+    # Temporary metrics
+    box_loss = tf.stop_gradient(0.05 * box_loss / self._iou_normalizer)
+
+    # Metric compute using done here to save time and resources.
+    sigmoid_conf = tf.stop_gradient(tf.sigmoid(pred_conf))
+    iou = tf.stop_gradient(iou)
+    avg_iou = loss_utils.average_iou(
+        loss_utils.apply_mask(tf.squeeze(ind_mask, axis=-1), iou))
+    avg_obj = loss_utils.average_iou(
+                    tf.squeeze(sigmoid_conf, axis=-1) * grid_mask)
+    return (loss, box_loss, conf_loss, class_loss, mean_loss,
+            tf.stop_gradient(avg_iou), tf.stop_gradient(avg_obj))
+
+  @abc.abstractmethod
+  def _build_per_path_attributes(self):
+    """Additional initialization required specifically for each unique YOLO 
+    loss version"""
+    ...
+
+  @abc.abstractmethod
+  def _compute_loss(self, true_counts, inds, y_true, boxes, classes, y_pred):
+    """The actual logic to apply to the raw model for optimization."""
+    ...
+
+  def post_path_aggregation(self, loss, ground_truths, predictions):
+    """This method allows for post processing of a loss value after the loss
+    has been aggregateda across all the FPN levels. The default behavior is to 
+    pass the loss through with no alterations."""
+    return loss
+
+  @abc.abstractmethod
+  def cross_replica_aggregation(self, loss, num_replicas_in_sync):
+    """This controls how the loss should be aggregated across replicas."""
+    ...
+
+
+@tf.custom_gradient
+def grad_sigmoid(values):
+  """
+  This function scales the gradient as if a signmoid was applied to the 
+  model output. This is used in the Darknet Loss when the choosen box type 
+  is the scaled coordinate type. This function is used to match the propagated 
+  gradient to match that of the Darkent Yolov4 model. 
+  """
+  # This is an identity operation that will
+  # allow us to add some steps to the back propagation.
+  def delta(dy):
+    # Darknet only propagtes sigmoids for the boxes
+    # under some conditions, so we need this to selectively
+    # add the sigmoid to the chain rule
+    t = tf.math.sigmoid(values)
+    return dy * t * (1 - t)
+
+  return values, delta
+
+
+class DarknetLoss(YoloLossBase):
+  """This class implements the full logic for the standard Yolo models."""
+
+  def _build_per_path_attributes(self):
+    """Paramterization of pair wise search and grid generators for box 
```

Comment:
Only one line description. details to be added after a blank line.

Reply:
addressed